packages/core/core/src/requests/AssetRequest.js

```diff
@@ -47,16 +48,16 @@ const type = 'asset_request';
 function getId(input: AssetRequestInput) {
   // eslint-disable-next-line no-unused-vars
   let {optionsRef, ...hashInput} = input;
-  return md5FromOrderedObject({
-    type,
-    filePath: input.filePath,
-    env: input.env.id,
-    isSource: input.isSource,
-    sideEffects: input.sideEffects,
-    code: input.code,
-    pipeline: input.pipeline,
-    query: input.query ? objectSortedEntries(input.query) : null,
-  });
+  return hashString(
+    type +
+      input.filePath +
+      input.env.id +
+      String(input.isSource) +
+      String(input.sideEffects) +
+      (input.code ?? '') +
+      (input.pipeline ?? '') +
```

Comment:
`code = "", pipeline = "foo"` and `code="foo", "pipeline = ""` would produce the same hash

Reply:
Good point.